crates/hyperswitch_connectors/src/connectors/redsys/transformers.rs

```diff
@@ -1,228 +1,1499 @@
+use base64::{engine::general_purpose::STANDARD, Engine};
 use common_enums::enums;
-use common_utils::types::StringMinorUnit;
+use common_utils::{ext_traits::ValueExt, types::StringMinorUnit};
+use error_stack::ResultExt;
+use hmac::{Hmac, Mac};
 use hyperswitch_domain_models::{
+    address::Address,
     payment_method_data::PaymentMethodData,
-    router_data::{ConnectorAuthType, RouterData},
-    router_flow_types::refunds::{Execute, RSync},
-    router_request_types::ResponseId,
-    router_response_types::{PaymentsResponseData, RefundsResponseData},
-    types::{PaymentsAuthorizeRouterData, RefundsRouterData},
+    router_data::{ConnectorAuthType, ErrorResponse, RouterData},
+    router_flow_types::refunds::Execute,
+    router_request_types::{
+        BrowserInformation, CompleteAuthorizeData, PaymentsAuthorizeData, PaymentsCancelData,
+        PaymentsCaptureData, PaymentsPreProcessingData, ResponseId,
+    },
+    router_response_types::{PaymentsResponseData, RedirectForm, RefundsResponseData},
+    types::{
+        PaymentsAuthorizeRouterData, PaymentsCancelRouterData, PaymentsCaptureRouterData,
+        PaymentsCompleteAuthorizeRouterData, PaymentsPreProcessingRouterData, RefundsRouterData,
+    },
 };
 use hyperswitch_interfaces::errors;
-use masking::Secret;
+use masking::{ExposeInterface, Secret};
+use openssl::symm::{encrypt, Cipher};
 use serde::{Deserialize, Serialize};
+use sha2::Sha256;
 
 use crate::{
     types::{RefundsResponseRouterData, ResponseRouterData},
-    utils::PaymentsAuthorizeRequestData,
+    utils::{
+        generate_12_digit_number, get_unimplemented_payment_method_error_message,
+        is_payment_failure, is_refund_failure, to_connector_meta, AddressDetailsData,
```

Comment:
Can we instead qualify usages of some of these functions?

`utils::generate_12_digit_number()` instead of directly calling `generate_12_digit_number()`?